jbehave-support-core/src/main/java/org/jbehavesupport/core/report/extension/ServerLogXmlReporterExtension.java

```diff
@@ -1,33 +1,67 @@
 package org.jbehavesupport.core.report.extension;
 
+import java.io.File;
+import java.io.IOException;
 import java.io.Writer;
-import java.util.ArrayList;
-import java.util.Arrays;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Date;
 import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
-import java.util.stream.Collectors;
 
+import lombok.Getter;
 import lombok.RequiredArgsConstructor;
-import org.jbehavesupport.core.report.ReportContext;
-import org.jbehavesupport.core.ssh.SshLog;
-import org.jbehavesupport.core.ssh.SshTemplate;
-
+import lombok.Setter;
+import org.apache.commons.collections4.keyvalue.MultiKey;
+import org.apache.commons.collections4.map.LRUMap;
+import org.apache.commons.collections4.map.MultiKeyMap;
 import org.apache.commons.lang3.exception.ExceptionUtils;
-import org.springframework.beans.factory.annotation.Qualifier;
-import org.springframework.beans.factory.config.BeanDefinition;
-import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
-import org.springframework.beans.factory.support.RootBeanDefinition;
+import org.apache.commons.io.FileUtils;
+import org.jbehavesupport.core.TestContext;
+import org.jbehavesupport.core.report.ReportContext;
+import org.jbehavesupport.core.internal.FileNameResolver;
+import org.jbehavesupport.core.report.ReportRenderingPhase;
+import org.jbehavesupport.core.ssh.SshReportType;
+import org.springframework.beans.factory.annotation.Value;
 
 @RequiredArgsConstructor
 public class ServerLogXmlReporterExtension extends AbstractXmlReporterExtension {
 
     private static final String SSH_XML_REPORTER_EXTENSION = "serverLog";
     private static final String LOG = "log";
+    private static final String FILE = "file";
+    private static final String TEXT = "text";
     private static final String SYSTEM = "system";
     private static final String FAIL = "fail";
+    private static final String FILE_NAME_PATTERN = "LOG_%s_%s.txt";
+    private static final Long MAX_LENGTH = 10000L;
+
+    private final TestContext testContext;
+    private final FileNameResolver fileNameResolver;
+    @Getter
+    @Setter
+    private SshReportType sshReportType;
 
-    private final ConfigurableListableBeanFactory beanFactory;
+    @Value("${ssh.reporting.logOnFailure:false}")
+    @Getter
+    private boolean loggingOnFailure;
+
+    @Value("${web.reporting.directory:./target/reports}")
+    private String logFileDirectory;
+
+    @Override
+    public ReportRenderingPhase getReportRenderingPhase() {
+        return ReportRenderingPhase.AFTER_STORY;
+    }
+
+    private MultiKeyMap<MultiKey, String> logContents = MultiKeyMap.multiKeyMap(new LRUMap());
+
+    public void registerLogContent(MultiKey multiKey, String logContent) {
+        if (logContents.containsKey(multiKey)) {
+            logContent = logContent.concat(logContents.get(multiKey));
```

Comment:
Is simple concat the way we want to go? I dont have anything against it, just asking and pointing it out.

Reply:
This is not popular anymore?
`logContent += logContents.get(multiKey);`